File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.java
import java.util.*;
class Solution {
    public int[] solution(int n, int[] numlist) {
        int[] answer = {};
        
        int index = 0;
        
        for(int i = 0; i < numlist.length; i++){
            int num = numlist[i];
            
            if(num % n == 0) {
                numlist[index] = num;
                index++;
            }
        }
        
        answer = Arrays.copyOfRange(numlist, 0, index);
        
        return answer;
    }
}

File: 프로그래머스/0/120905. n의 배수 고르기/n의 배수 고르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, int numlist[], size_t numlist_len) {
    int* answer = (int*)malloc(numlist_len * sizeof(int));
    int max_size = 0;
    int idx = 0;
    
    for(int i = 0; i < numlist_len; i++){
        if(numlist[i] % n == 0){
            answer[idx++] = numlist[i];
            max_size++;
        }
    }
    
    answer = realloc(answer, max_size * sizeof(int));
    
    return answer;
}